code/game/machinery/doors/windowdoor.dm

```diff
@@ -215,6 +227,16 @@
 	health = 300.0 //Stronger doors for prison (regular window door health is 150)
 
 
+/obj/machinery/door/window/brigdoor/allowed(mob/M)
+	if(inoperable()) // Brigdoors are the exception to the "fail open" windoor - they lock closed
+		return FALSE
+	. = ..()
+
+/obj/machinery/door/window/brigdoor/power_change()
+	..()
+	if((stat & NOPOWER) && !density)
+		close(TRUE)
```

Comment:
Maybe add a message (perhaps in limited range) like "as the power shuts down the windoor closes".

Reply:
Eh, that message doesn't exist for airlocks opening/closing on power loss and in places like the brig it'd just spam the chat log.

Reply:
True. I wonder if showing a message (for _all_ doors that do this) in at least a very small range (maybe 2/3 tiles) would be meaningful, but I guess power outage is a visible enough even that it doesn't make much sense. Though I personally was quite surprised when I saw doors open without power, lol.